README.md

```diff
@@ -1547,6 +1340,238 @@ def train(images, classes, epochs, model, device, num_classes=10, batch_size=32)
 train(images, classes, num_epochs, model, device, num_classes = num_classes, batch_size = batch_size)
 ```
 
+</details>
+</details>
+
+</details>
+
+</details>
+
+## Ivy in Depth
```

Comment:
I think we should have a very brief introduction before the collapsed sections here

Reply:
would you like to have this on a new PR? just to minimise the scope here and move faster?

Reply:
I feel the collpased sections are too isolated at the moment. Specially being after all the other collapsed sections, I think we should have the introductions before merging these changes to give this section more structure

Reply:
Thanks for that! added one line explainer.